app/api/Spaces.scala

```diff
@@ -303,14 +303,26 @@ class Spaces @Inject()(spaces: SpaceService,
 
 
   def listDatasets(spaceId: UUID, limit: Integer) = PermissionAction(Permission.ViewSpace, Some(ResourceRef(ResourceRef.space, spaceId))) { implicit request =>
-    val datasetList = datasets.listSpace(limit, spaceId.stringify)
-    Ok(toJson(datasetList))
+    spaces.get(spaceId) match {
+      case Some(space) =>{
+        val datasetList = datasets.listSpace(limit, spaceId.stringify)
+        Ok(toJson(datasetList))
+      }
+      case None => NotFound(s"Space $spaceId not found.")
+    }
+
   }
 
 
   def listCollections(spaceId: UUID, limit: Integer) = PermissionAction(Permission.ViewSpace, Some(ResourceRef(ResourceRef.space, spaceId))) { implicit request =>
-    val collectionList = collectionService.listSpace(limit, spaceId.stringify)
-    Ok(toJson(collectionList))
+    spaces.get(spaceId) match {
+      case Some(space) =>  {
+        val collectionList = collectionService.listSpace(limit, spaceId.stringify)
+        Ok(toJson(collectionList))
+      }
+      case None => NotFound(s"Space $spaceId not found.")
+    }
```

Comment:
in here and the above method `listDatasets` i noticed that even if a not-existing space Id get passed in, it still return 200 with an empty list. So I added a layer of check if space exist or not. 
But I'm not sure if this is against the original design or not. Let me know and I can change them back.

Reply:
I agree - I think that handling it the way that you have is a great idea.. previously, users might have been able to pass in bad input without realizing (e.g. a `spaceId` that has been deleted), and might be confused why the returned results where empty.

This is often an API pattern used for lists that have multiple optional filters, so that if one filter is invalid it doesn't necessarily ruin the entire query. If the `spaceId` were an optional parameter, then the current implementation might be more favorable/flexible. Since `spaceId` appears to be required here, I think it makes sense to validate the inputs somewhere :+1:

It looks like `MongoDBCollectionService` does not already perform this type of validation at the lower level, so it might be best to handle it here in the API. I think that typically the "service" layer is very thin, with the API performing the actual validation, but that might be a bigger pattern question for @lmarini or @robkooper to weigh in on.

Reply:
Yeah, this pattern is mostly what we  have been doing, with the Action handling the check. It would be nice if the CollectionService would throw specific exceptions but unfortunately it does not. LGTM as well.

Reply:
> Please update changelog. Thank you!

done~